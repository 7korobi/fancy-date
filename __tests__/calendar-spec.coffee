{ FictionalDate, to_timer, to_msec } = require "../lib/index.min"

g = FictionalDate.Gregorian

describe "define", =>
  test 'data', =>
    expect g.dic
    .toMatchSnapshot()
    expect g.calc.msec.period
    .toEqual 12622780800000
    expect g.calc.msec_max.period
    .toEqual 12622780800000
    expect g.calc.msec_min.period
    .toEqual 12622780800000
    expect g.table.msec.year[-1..]
    .toEqual [12622780800000]
    return
  return

describe "standard use", =>
  test 'format', =>
    expect g.format 100000000000000
    .toEqual "西暦5138年霜月16日(水)18時"

    expect g.format 10000000000000
    .toEqual "西暦2286年霜月21日(日)2時"

    expect g.format 1000000000000
    .toEqual "西暦2001年長月9日(日)10時"

    expect g.format 100000000000
    .toEqual "西暦1973年弥生3日(土)18時"

    expect g.format 10000000000
    .toEqual "西暦1970年卯月27日(月)2時"

    expect g.format 0
    .toEqual "西暦1970年睦月1日(木)9時"
    return
  return
