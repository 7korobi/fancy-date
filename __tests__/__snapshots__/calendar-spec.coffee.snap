// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`define data 1`] = `
Object {
  "axtial_tilt": 23.4,
  "era": Array [
    "西暦",
    "紀元前",
  ],
  "geo": Array [
    35,
    135,
  ],
  "hours": Array [
    "0時",
    "1時",
    "2時",
    "3時",
    "4時",
    "5時",
    "6時",
    "7時",
    "8時",
    "9時",
    "10時",
    "11時",
    "12時",
    "13時",
    "14時",
    "15時",
    "16時",
    "17時",
    "18時",
    "19時",
    "20時",
    "21時",
    "22時",
    "23時",
  ],
  "lat": 35,
  "leaps": Array [
    4,
    100,
    400,
  ],
  "lng": 135,
  "minutes": Array [
    "0分",
    "1分",
    "2分",
    "3分",
    "4分",
    "5分",
    "6分",
    "7分",
    "8分",
    "9分",
    "10分",
    "11分",
    "12分",
    "13分",
    "14分",
    "15分",
    "16分",
    "17分",
    "18分",
    "19分",
    "20分",
    "21分",
    "22分",
    "23分",
    "24分",
    "25分",
    "26分",
    "27分",
    "28分",
    "29分",
    "30分",
    "31分",
    "32分",
    "33分",
    "34分",
    "35分",
    "36分",
    "37分",
    "38分",
    "39分",
    "40分",
    "41分",
    "42分",
    "43分",
    "44分",
    "45分",
    "46分",
    "47分",
    "48分",
    "49分",
    "50分",
    "51分",
    "52分",
    "53分",
    "54分",
    "55分",
    "56分",
    "57分",
    "58分",
    "59分",
  ],
  "month_ranges": Array [
    31,
    0,
    31,
    30,
    31,
    30,
    31,
    31,
    30,
    31,
    30,
    31,
  ],
  "month_sum": 337,
  "months": Array [
    "睦月",
    "如月",
    "弥生",
    "卯月",
    "皐月",
    "水無月",
    "文月",
    "葉月",
    "長月",
    "神無月",
    "霜月",
    "師走",
  ],
  "moon_idx": 27,
  "moons": Array [
    "朔",
    "既朔",
    "三日月",
    "上弦",
    "上弦",
    "上弦",
    "上弦",
    "上弦",
    "上弦",
    "上弦",
    "上弦",
    "上弦",
    "十三夜",
    "小望月",
    "満月",
    "十六夜",
    "立待月",
    "居待月",
    "臥待月",
    "更待月",
    "下限",
    "下限",
    "下限",
    "下限",
    "下限",
    "下限",
    "下限",
    "下限",
    "晦",
    "晦",
  ],
  "seconds": Array [
    "0秒",
    "1秒",
    "2秒",
    "3秒",
    "4秒",
    "5秒",
    "6秒",
    "7秒",
    "8秒",
    "9秒",
    "10秒",
    "11秒",
    "12秒",
    "13秒",
    "14秒",
    "15秒",
    "16秒",
    "17秒",
    "18秒",
    "19秒",
    "20秒",
    "21秒",
    "22秒",
    "23秒",
    "24秒",
    "25秒",
    "26秒",
    "27秒",
    "28秒",
    "29秒",
    "30秒",
    "31秒",
    "32秒",
    "33秒",
    "34秒",
    "35秒",
    "36秒",
    "37秒",
    "38秒",
    "39秒",
    "40秒",
    "41秒",
    "42秒",
    "43秒",
    "44秒",
    "45秒",
    "46秒",
    "47秒",
    "48秒",
    "49秒",
    "50秒",
    "51秒",
    "52秒",
    "53秒",
    "54秒",
    "55秒",
    "56秒",
    "57秒",
    "58秒",
    "59秒",
  ],
  "start": "1970年1月1日(木)0時0分0秒",
  "start_at": 0,
  "tz_offset": 32400000,
  "weeks": Array [
    "日",
    "月",
    "火",
    "水",
    "木",
    "金",
    "土",
  ],
}
`;
